test/framework/easyblock.py

```diff
@@ -1411,6 +1412,51 @@ def test_this_is_easybuild(self):
         self.assertTrue(isinstance(get_git_revision(), str))
         self.assertTrue(isinstance(this_is_easybuild(), str))
 
+    def test_stale_module_caches(self):
+        """Test whether module caches are reset between builds."""
+
+        ec1 = os.path.join(self.test_prefix, 'one.eb')
+        ec1_txt = '\n'.join([
+            "easyblock = 'Toolchain'",
+            "name = 'one'",
+            "version = '1.0.2'",
+            "homepage = 'https://example.com'",
+            "description = '1st test easyconfig'",
+            "toolchain = {'name': 'dummy', 'version': ''}",
+        ])
+        write_file(ec1, ec1_txt)
+
+        # key aspect here is that two/2.0 depends on one/1.0 (which is an alias for one/1.0.2)
+        ec2 = os.path.join(self.test_prefix, 'two.eb')
+        ec2_txt = '\n'.join([
+            "easyblock = 'Toolchain'",
+            "name = 'two'",
+            "version = '2.0'",
+            "toolchain = {'name': 'dummy', 'version': ''}",
+            "homepage = 'https://example.com'",
+            "description = '2nd test easyconfig'",
+            "dependencies = [('one', '1.0')]",
+        ])
+        write_file(ec2, ec2_txt)
+
+        # populate modules avail/show cache with result for "show one/1.0" when it doesn't exist yet
+        # need to make sure we use same $MODULEPATH value as the one that is in place during build
+        moddir = os.path.join(self.test_installpath, 'modules', 'all')
+        self.modtool.use(moddir)
+        self.assertFalse(self.modtool.exist(['one/1.0'])[0])
+
+        # add .modulerc to install version alias one/1.0 for one/1.0.2
+        # this makes cached result for "show one/1.0" incorrect as soon as one/1.0.2 is installed via one.eb
+        modgen = module_generator(None)
+        module_version_spec = {'modname': 'one/1.0.2', 'sym_version': '1.0', 'version': '1.0.2'}
+        modulerc_txt = modgen.modulerc(module_version=module_version_spec)
+        one_moddir = os.path.join(self.test_installpath, 'modules', 'all', 'one')
+        write_file(os.path.join(one_moddir, '.modulerc'), modulerc_txt)
+
```

Comment:
What happens if you repeat 1446 at this point (1.0.2 doesn't exist yet but the rc file does)?

Reply:
Well, it won't make a difference because of the modules cache (the result for `avail one/1.0` & `show one/1.0` is cached).
And even if we clear the cache, it will still fail, since the underlying module doesn't exist yet (I'll add that to the test).